org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/ClassFile.java

```diff
@@ -3959,6 +3970,55 @@ private int addClassDescBootstrap(int localContentsOffset, TypeBinding type, Map
 
 		return localContentsOffset;
 	}
+	private int addPrimitiveClassBootstrap(int localContentsOffset, Map<String, Integer> fPtr) {
+		final int contentsEntries = 4;
+		int idx = fPtr.get(CONSTANT_BOOTSTRAP__PRIMITIVE_CLASS);
```

Comment:
The tokens  `PRIMITIVE_CLASS__TOKEN` and `GET_STATIC_FINAL__TOKEN`  can be refactored to descriptors to the end that this method and `addGetBooleanConstantsBootstrap` can be folded into one.

Reply:
done in [a62bba7](https://github.com/eclipse-jdt/eclipse.jdt.core/pull/2866/commits/a62bba785f264a1f3f884aefa2e1bf416868fc4c)